### PR TITLE
APR/CAPER 2024 v1.2 updates

### PR DESCRIPTION
**Pull request by eanders**
## Description

This updates the APR and CAPER to the newest version of the spec and updates the definition of "parenting youth" to match the response from the Data Lab.

## Type of change
- [x] Bug fix
- [x] New feature (adds functionality)
- [x] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: app/models/concerns/hud_reports/households.rb [hunk 1]
```diff
@@ -202,8 +202,13 @@ def hoh_age(household_id, date)
       household_adults(universe_client).map { |member| member['source_client_id'] }
     end
 
+    # Per HUD:
+    # https://airtable.com/appFAz3WpgFmIJMm6/shr8TvO6KfAZ3mOJd/tblYhwasMJptw5fjj/viw7VMUmDdyDL70a7/recCDGtYIVXlTmAvk
+    # . The glossary does not make reference to the "relationship to head of household" required to include a youth in this category. Q27b ("Parenting Youth") is a bit clearer in the following language: "Report all heads of household plus all adults (age 18 – 24) in the household in column B according to the age of the head of household (age < 18 on line 2, or 18-24 on line 3). Include all adults in the household regardless of [relationship to head of household],"
     private def youth_parent?(universe_client)
-      universe_client.head_of_household && only_youth?(universe_client) && youth_children?(universe_client)
+      age = universe_client
```
Comment (awisema) [inline app/models/concerns/hud_reports/households.rb]: I think something got lost here
```suggestion
      age = universe_client.age
```
?

**Review by awisema — CHANGES_REQUESTED** [review]

File: app/models/concerns/hud_reports/households.rb (hunk 1 above)
Comment (eanders) [reply app/models/concerns/hud_reports/households.rb]: Whops!

File: app/models/grda_warehouse/ch_enrollment.rb [hunk 2]
```diff
@@ -301,10 +313,27 @@ def self.homeless_duration_sufficient(enrollment, date: enrollment.EntryDate)
     # Add steps for lines 15-16 and lines 22-23
     def self.length_of_stay_previous_sufficient(enrollment)
       steps = []
-      steps.push({ result: is_no?(enrollment.LOSUnderThreshold) || :continue, display_value: enrollment.LOSUnderThreshold })
+      value = enrollment.LOSUnderThreshold
+      result = if is_no?(value)
+        is_no?(value)
```
Comment (awisema) [inline app/models/grda_warehouse/ch_enrollment.rb]: Maybe
```suggestion
       :no
```
To be consistent?

File: app/models/grda_warehouse/ch_enrollment.rb [hunk 3]
```diff
@@ -301,10 +313,27 @@ def self.homeless_duration_sufficient(enrollment, date: enrollment.EntryDate)
...
+        :missing
+      end
+
+      steps.push({ result: result, display_value: enrollment.LOSUnderThreshold })
       return steps if terminating?(steps.last[:result])
 
-      steps.push({ result: is_no?(enrollment.PreviousStreetESSH) || :continue, display_value: enrollment.PreviousStreetESSH })
+      value = enrollment.PreviousStreetESSH
+      result = if is_no?(value)
+        is_no?(value)
```
Comment (awisema) [inline app/models/grda_warehouse/ch_enrollment.rb]:
```suggestion
        :no
```

**Review by awisema — APPROVED** [review]

**Pull request merged**